cocotb/triggers.py

```diff
@@ -52,7 +52,36 @@
 class TriggerException(Exception):
     pass
 
-class Trigger(_py_compat.with_metaclass(abc.ABCMeta)):
+
+class TriggerHelper(object):
+    """A base class with commen methods for event type classes"""
+
+    @decorators.coroutine
+    def aslongas(self, condition, readonly=True):
+        """yield itself as long as condition is ``True``"""
+        yield WaitCondition(lambda: not condition(), event=self, readonly=readonly, before=False, after=False)
+
+    @decorators.coroutine
+    def aslongas_after(self, condition, readonly=True):
+        """yield itself as long as condition is ``True`` with an extra yield afterwards.
+
+        No extra ReadOnly will be done after the extra yield."""
+        yield WaitCondition(lambda: not condition(), event=self, readonly=readonly, before=False, after=True)
+
+    @decorators.coroutine
+    def until(self, condition, readonly=True):
+        """yield itself until condition is ``True``"""
+        yield WaitCondition(condition, event=self, readonly=readonly, before=True, after=False)
+
+    @decorators.coroutine
```

Comment:
You can remove this decorator, and replace the `yield` with `return` below, same for the others

Reply:
Done